src/app/core/services/rundown-timing-context-state.service.ts

```diff
@@ -0,0 +1,187 @@
+  ) {
+    this.startLowResolutionTimer()
+  }
+
+  public async subscribeToRundownTimingContext(rundownId: string): Promise<Observable<RundownTimingContext>> {
+    const rundownTimingContextSubject: BehaviorSubject<RundownTimingContext> = await this.createRundownTimingContextSubject(rundownId)
+    return rundownTimingContextSubject.asObservable()
+  }
+
+  private startHighResolutionTimer(): void {
+    this.stopTimeResolutionTimer()
+    this.isHighResolutionTimer = true
+    this.timeResolutionTimerId = setInterval(this.onTimeTick.bind(this), HIGH_RESOLUTION_INTERVAL_DURATION_IN_MS)
+  }
+
+  private onTimeTick(): void {
+    this.rundowns.forEach(rundown => {
+      const rundownTimingContextSubject: BehaviorSubject<RundownTimingContext> | undefined = this.getRundownTimingContextSubject(rundown.id)
+      if (!rundownTimingContextSubject) {
+        return
+      }
+      const currentEpochTime: number = Date.now()
+      const previousRundownTimingContext: RundownTimingContext = rundownTimingContextSubject.value
+      const expectedStartEpochTimeForRundown: number = this.rundownTimingService.getExpectedStartEpochTimeForRundown(
+        rundown,
+        previousRundownTimingContext.expectedDurationInMsForRundown,
+        currentEpochTime
+      )
+      const expectedEndEpochTimeForRundown: number = this.rundownTimingService.getExpectedEndEpochTimeForRundown(rundown, previousRundownTimingContext.expectedDurationInMsForRundown, currentEpochTime)
+      const playedDurationInMsForOnAirPart: number = this.rundownTimingService.getPlayedDurationInMsForOnAirPart(rundown, currentEpochTime)
+      const playedDurationInMsForOnAirSegment: number = this.rundownTimingService.getPlayedDurationInMsForOnAirSegment(rundown, currentEpochTime)
+      const durationInMsSpentInOnAirSegment: number = this.rundownTimingService.getDurationInMsSpentInOnAirSegment(rundown, currentEpochTime)
+      const remainingDurationInMs: number = this.rundownTimingService.getRemainingDurationInMsForRundown(
+        rundown,
+        previousRundownTimingContext.expectedDurationsInMsForSegments,
+        playedDurationInMsForOnAirSegment
+      )
+      const rundownTimingContext: RundownTimingContext = {
+        ...rundownTimingContextSubject.value,
+        currentEpochTime,
+        expectedStartEpochTimeForRundown,
+        expectedEndEpochTimeForRundown,
+        playedDurationInMsForOnAirPart,
+        playedDurationInMsForOnAirSegment,
+        durationInMsSpentInOnAirSegment,
+        remainingDurationInMsForRundown: remainingDurationInMs,
+      }
+      rundownTimingContextSubject.next(rundownTimingContext)
+    })
+
+    const hasActiveRundown: boolean = [...this.rundowns.values()].some(rundown => rundown.isActive)
+    if (hasActiveRundown && !this.isHighResolutionTimer) {
+      this.startHighResolutionTimer()
+    } else if (!hasActiveRundown && this.isHighResolutionTimer) {
+      this.startLowResolutionTimer()
+    }
```

Comment:
Do you really need to check for `isHighResolutionTimer`?

Reply:
Since this is using an interval as the timer, yes. But I have updated the time tick scheduling to use timeouts instead. It becomes a bit more imprecise, but removes the need for the state variable `isHighResolutionTimer`.